python-packaging/execute-code.md

```diff
@@ -0,0 +1,167 @@
+
+def shiny_hello():
+    print("\N{Sparkles} Hello from Python \N{Sparkles}")
+
+shiny_hello()
+```
+
+```bash
+my_program.py
+# ✨ Hello from Python ✨
+```
+
+:::{tip}
+Shebangs are a feature of POSIX. POSIX represents some level of compatibility between systems.
+Linux, macOS, all BSDs, and many other operating systems are fully- or mostly-POSIX compliant.
+
+Windows is not natively POSIX compliant. However, some "modes" inside of Windows are, such as WSL
+(Windows Subsystem for Linux), gitbash, or some VSCode terminals.
+:::
+
+### Windows executables
+
+If your Windows machine has Python registered as the default application associated with `.py` files, then any Python
+scripts can be run as commands. However, only one Python can be registered at a time, so all Python scripts run this
+way will use the same Python environment. While all Python files should end in a `.py`, this naming is necessary for
+Windows to associate the file with Python, as opposed to on Linux where `.py` is a convention and the shebang associates
+the file with Python.
+
+Additionally, most Windows Python installs come with the [Python Launcher](https://docs.python.org/3/using/windows.html#python-launcher-for-windows)
+which, in addition to allowing specifying the version of Python, can also read shebang lines and emulate some of that behavior.
+This allows for shebang lines to be re-used between Linux, macOS, and Windows systems. However, on Windows the command must still
+be prefaced with another command (`py`).
+
+```bash
+py my_program.py
+# ✨ Hello from Python ✨
+```
+
+:::{tip}
+While there is no in-source format that can tell Windows what to do with a Python code file, executing a
+Python file with a shebang on Windows also does not cause any issues. Python just sees the whole line as
+a comment and ignores it! So even if you develop on Windows, it may be a good idea to add the shebang as
+the first line of your scripts, so that colleagues on different systems can also run it.
+:::
+
+## Executable comparisons
+
+### Pros of passing a file to `python`:
+- don't need execute permissions
+- works for every system
+- explicit about what you expect to happen
+
+### Pros of inserting a shebang to the file:
+- file is associated with specific python
+  - don't have to remember which
+- don't have to use the `python` command
+  - don't have to even remember it is a Python script
+
+
+# Share Code
+
+## Execute a python package
+
+In [Code Workflow Logic][Code Workflow Logic] you learned of the two primary ways to execute a stand-alone Python script.
+There are two other ways to execute Python as commands, both of which work for code that has been formatted as a package.
+
+### Entrypoints
+
+There is a special `entrypoint` a package can specify in its configuration which will direct installers to create an
+executable command. Entrypoints are a general purpose plug-in system for Python packages, but the
```

Comment:
this sentence and something like "that you can run from your shell anywhere, without needing to refer to the file specifically."

And then maybe a demo - 

"that's how people make it possible to do things like this:

```shell
$ pip install my_package
$ my_package
```
"

or something basic like that - they will have probably seen that before and maybe wondered how to do that, so i think a concrete example to let them know that "this is where we learn to do this cool thing" would be nice :)